File: services/gateway/src/routes/projects.js
'use strict';

const express = require('express');
const { getApiKey } = require('@ai-fleet/shared/store');
const {
  getProjects,
  getProjectMilestones,
  getTeams,
} = require('@ai-fleet/shared/linear');
const { asyncHandler } = require('@ai-fleet/shared/util');

const router = express.Router();

// GET /api/projects — all Linear projects.
router.get(
  '/',
  asyncHandler(async (req, res) => {
    const projects = await getProjects(getApiKey());
    res.json({ projects });
  })
);

// GET /api/projects/teams — teams (used when creating a project for a business).
router.get(
  '/teams',
  asyncHandler(async (req, res) => {
    const teams = await getTeams(getApiKey());
    res.json({ teams });
  })
);

// GET /api/projects/:id/milestones — the milestone planning view.
router.get(
  '/:id/milestones',
  asyncHandler(async (req, res) => {
    const { project, milestones } = await getProjectMilestones(getApiKey(), req.params.id);
    res.json({ project, milestones });
  })
);

module.exports = router;

File: ai_fleet/services/gateway/routes/projects.py
"""Projects routes (port of services/gateway/src/routes/projects.js).

Read-only Linear project surface: list projects, list teams (used when creating a
project for a business), and the milestone planning view for one project.
"""

from __future__ import annotations

from fastapi import APIRouter

from ai_fleet import store, linear

router = APIRouter()


# GET /api/projects — all Linear projects.
@router.get("")
@router.get("/")
async def list_projects():
    projects = await linear.get_projects(store.get_api_key())
    return {"projects": projects}


# GET /api/projects/teams — teams (used when creating a project for a business).
@router.get("/teams")
async def list_teams():
    teams = await linear.get_teams(store.get_api_key())
    return {"teams": teams}


# GET /api/projects/:id/milestones — the milestone planning view.
@router.get("/{id}/milestones")
async def project_milestones(id: str):
    result = await linear.get_project_milestones(store.get_api_key(), id)
    return {"project": result["project"], "milestones": result["milestones"]}
